src/renderer/sw_engine/tvgSwStroke.cpp

```diff
@@ -370,12 +370,13 @@ void _firstSubPath(SwStroke& stroke, SwFixed startAngle, SwFixed lineLength)
 }
 
 
-static void _lineTo(SwStroke& stroke, const SwPoint& to)
+static bool _lineTo(SwStroke& stroke, const SwPoint& to)
 {
     auto delta = to - stroke.center;
 
     //a zero-length lineto is a no-op; avoid creating a spurious corner
-    if (delta.zero()) return;
+    if (delta.zero())
```

Comment:
We can provide the de-coupled logic like this. (same to cubic to):
```
    if (delta.zero()) {
        //Round and square caps are expected to be drawn as a dot even for zero-length lines.
        if (stroke.firstPt && stroke.cap != StrokeCap::Butt) _firstSubPath(stroke, 0, 0);
        return;
    }
```

Reply:
Very good idea. Thank you.